### PR TITLE
fix: show empty pop when no result is found.

### LINKED ISSUE #14678: [Bug]: Phone & currency Input widget Popup breaks on wrong search text in popup filter
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When we do quick search on popup that appears to change country code/currency, the popup breaks when we do wrong search instead of showing 'No results'

https://www.loom.com/share/e876c382aedf47eb9b349d9ef7b19245

### Steps To Reproduce

1. Add Phone Input/Currency Input Widget onto canvas
2. Enable Change Country Code/Currency option from property pane
3. Now search with wrong input in the popup of change country code/currency
4. Observe that popup disappears as soon as we add wrong text in the search input

### Public Sample App

_No response_

### Version

Cloud/Production

### PR DESCRIPTION
**Pull request by yaldram**
## Description
- For the `currency` and `phoneinput` widgets when the search result does not match the options the search pop is closed.
- Added another prop called `showEmptyOptions` in the `Dropdown` to handle the above scenario

Fixes #14678 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- Tested Manually on the canvas

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/client/src/components/ads/Dropdown.tsx [hunk 1]
```diff
@@ -726,7 +734,7 @@ export function RenderDropdownOptions(props: DropdownOptionsProps) {
   };
   const theme = useTheme() as Theme;
 
-  if (!options.length) return null;
+  if (!options.length && !showEmptyOptions) return null;
```
Comment (yaldram) [inline app/client/src/components/ads/Dropdown.tsx:737]: @akash-codemonk the last changes were done by you here, please leave your review, thanks.

**Review by akash-codemonk — APPROVED** [review]

**Comment by akash-codemonk** [comment]
Do we want to add a cypress test for this?

**Comment by yaldram** [comment]
@akash-codemonk Yes I will add cypress tests for both the widgets.

**Review by sbalaji1192 — APPROVED** [review]

**Pull request merged**